lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -229,22 +242,47 @@ def _get_hostname(self, item):
             :param item: A host response from GCP
             :return the hostname of this instance
         '''
+        hostname_ordering = ['public_ip', 'private_ip', 'name']
+        if self.get_option('hostnames'):
+            hostname_ordering = self.get_option('hostnames')
+
+        for order in hostname_ordering:
+            name = None
+            if order == 'public_ip':
+                name = self._get_publicip(item)
+            elif order == 'private_ip':
+                name = self._get_privateip(item)
+            elif order == 'name':
+                name = item['name']
+            else:
+                raise AnsibleParserError("%s is not a valid hostname precedent" % order)
+
+            if name:
+                return name
```

Comment:
I think this should have an `else` on the `for` to raise an exception if none of the user-provided hostname preferences match, so we don't add hosts with the hostname `None` which would be the default return.

Reply:
That's a great note. Made that change.